core/bin/block_reverter/src/main.rs

```diff
@@ -126,111 +132,112 @@ async fn main() -> anyhow::Result<()> {
+        .basic_witness_input_producer_config
+        .clone()
+        .context("Failed to find eth config")?;
+    let contracts = read_yaml_repr::<zksync_protobuf_config::proto::contracts::Contracts>(
+        &opts
+            .contracts_config_path
+            .context("Missing contracts config")?,
+    )
+    .context("failed decoding contracts YAML config")?;
+
+    let secrets_config = read_yaml_repr::<zksync_protobuf_config::proto::secrets::Secrets>(
+        &opts.secrets_path.context("Missing secrets config")?,
+    )
+    .context("failed decoding secrets YAML config")?;
 
     let default_priority_fee_per_gas = eth_sender
         .gas_adjuster
         .context("gas_adjuster")?
         .default_priority_fee_per_gas;
 
-    let database_secrets = match &secrets_config {
-        Some(secrets_config) => secrets_config
-            .database
-            .clone()
-            .context("Failed to find database config")?,
-        None => DatabaseSecrets::from_env().context("DatabaseSecrets::from_env()")?,
-    };
-    let l1_secrets = match &secrets_config {
-        Some(secrets_config) => secrets_config
-            .l1
-            .clone()
-            .context("Failed to find l1 config")?,
-        None => L1Secrets::from_env().context("L1Secrets::from_env()")?,
-    };
-    let postgres_config = match &general_config {
-        Some(general_config) => general_config
-            .postgres_config
-            .clone()
-            .context("Failed to find postgres config")?,
-        None => PostgresConfig::from_env().context("PostgresConfig::from_env()")?,
-    };
-    let zksync_network_id = match &genesis_config {
-        Some(genesis_config) => genesis_config.l2_chain_id,
-        None => {
-            NetworkConfig::from_env()
-                .context("NetworkConfig::from_env()")?
-                .zksync_network_id
-        }
+    let database_secrets = secrets_config
+        .database
+        .clone()
+        .context("Failed to find database config")?;
+
+    let l1_secrets = secrets_config
+        .l1
+        .clone()
+        .context("Failed to find l1 config")?;
+
+    let postgres_config = general_config
+        .postgres_config
+        .clone()
+        .context("Failed to find postgres config")?;
+
+    let zksync_network_id = genesis_config.l2_chain_id;
+
+    let settlement_mode = general_config
+        .eth
+        .context("Missing `eth` config")?
+        .gas_adjuster
+        .context("Missing `gas_adjuster` config")?
+        .settlement_mode;
+
+    let (sl_rpc_url, sl_diamond_proxy, sl_validator_timelock) = if settlement_mode.is_gateway() {
+        let gateway_chain_config: GatewayChainConfig =
+            read_yaml_repr::<zksync_protobuf_config::proto::gateway::GatewayChainConfig>(
+                &opts
+                    .gateway_chain_path
+                    .context("Genesis config path not provided")?,
+            )
+            .context("failed decoding genesis YAML config")?;
```

Comment:
By how much would it help you? I can ofc redo to make the thing a part of the `ContractsConfig`, but it will be migration of a "gateway config file" into a "gateway config struct" that is a part of `ContractsConfig`

Reply:
Discussed in private. It was decided that merge + fix later is fine